### ISSUE TITLE
flutter debug can't work in version 0.40.4

### ISSUE THREAD

**Issue by DevinFu** [getcursor/cursor#1829]
I am a Flutter developer. Today, I updated to the latest version of Cursor, with the following version information:
Version: 0.40.4
VSCode Version: 1.91.1
Commit: 63dae81aea2c3fc079420fe72578a8eef26ba870
Date: 2024-09-05T02:52:05.660Z
Electron: 29.4.0
ElectronBuildId: undefined
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin x64 22.6.0
After the update, I noticed that the time it takes for Flutter attach to connect successfully is much longer than before. Even worse, after a successful connection, when I modify the code and save it, the app freezes after hot reload is triggered. The app only responds to operations again after disconnecting the debug session.
In other words, after updating to the new version, the debugging functionality for Flutter is essentially completely unusable. I don't know what you changed in this version, but this is clearly a major issue. I hope you can fix it as soon as possible. Thank you!

**Comment by DevinFu** [comment]
Additional information:
Flutter plugin version: v3.96.0
Dart plugin version: v3.96.0
